extensions/amp-story/1.0/amp-story.js

```diff
@@ -994,7 +994,9 @@ export class AmpStory extends AMP.BaseElement {
           oldPage.setState(PageState.NOT_ACTIVE);
 
           // Indication that this should be offscreen to left in desktop view.
-          setAttributeInMutate(oldPage, Attributes.VISITED);
+          this.getPageIndex(oldPage) < pageIndex ?
+            setAttributeInMutate(oldPage, Attributes.VISITED) :
+            removeAttributeInMutate(oldPage, Attributes.VISITED);
```

Comment:
If `Attributes.VISITED` is just deciding whether the pages are hidden off-screen to the right or left, maybe we should change the name. e.g. `Attributes.HIDE_LEFT`.

Also, have you tried if this works with RTL?

Reply:
See new commit :))